app/Http/Controllers/ChatController.php

```diff
@@ -31,17 +31,29 @@ public function index()
         $user = auth()->user();
         Chat::ack($user);
 
+        $params = get_params(request()->all(), null, [
+            'channel_id:int',
+            'sendto:int',
+        ], ['null_missing' => true]);
+
         // rejoin any existing channel first, otherwise it'll be missing from the preload later.
-        $targetUser = User::lookup(get_int(request('sendto')), 'id');
-        if ($targetUser !== null) {
-            $channel = Channel::findPM($targetUser, $user);
-            $channel?->addUser($user);
-
-            $sendToJson = [
-                'can_message_error' => ($channel?->checkCanMessage($user) ?? priv_check('ChatPmStart', $targetUser))->message(),
-                'channel_id' => $channel?->getKey(),
-                'target' => json_item($targetUser, 'UserCompact'),
-            ];
+        if ($params['sendto'] !== null) {
+            $targetUser = User::lookup($params['sendto'], 'id');
+            if ($targetUser !== null) {
+                $channel = Channel::findPM($targetUser, $user);
+                $channel?->addUser($user);
+
+                $sendToJson = [
+                    'can_message_error' => ($channel?->checkCanMessage($user) ?? priv_check('ChatPmStart', $targetUser))->message(),
+                    'channel_id' => $channel?->getKey(),
+                    'target' => json_item($targetUser, 'UserCompact'),
+                ];
+            }
+        } elseif ($params['channel_id'] !== null) {
+            $channel = Channel::find($params['channel_id']);
+            if ($channel?->isHideable() ?? false) {
+                $channel->addUser($user);
```

Comment:
I think this is missing permission check? If anything, this should be queried from UserChannel instead so the `addUser` action will be limited to unhiding.